KeenClient/KeenClient.m

```diff
@@ -291,78 +228,140 @@ + (BOOL)validateKey:(NSString *)key {
     return [KeenClient validateProjectID:key];
 }
 
-- (id)initWithProjectID:(NSString *)projectID andWriteKey:(NSString *)writeKey andReadKey:(NSString *)readKey {
+- (instancetype)initWithNetwork:(KIONetwork*)network
+                       andStore:(KIODBStore*)store
+                    andUploader:(KIOUploader*)uploader {
+    self = [super init];
+
+    if (nil != self) {
+        // log the current version number
+        if ([KeenClient isLoggingEnabled]) {
```

Comment:
I agree it can go away. This was an instance where I was just preserving what was already there to manage the scope of the change, but I'm ok with getting rid of it.